Volodin Evgeniy. Lab №1. Option №1.

evldn opened the pull request:
In this lab, a plugin was implemented that prints the names of all classes (structures) and their fields.
Example:
```C++ 
class MyClass {
    int a;
    float b;
};
```
Result:
```
MyClass
 |_a
 |_b`
```

evldn committed:
Task is ready!

evldn committed:
Update!

evldn committed:
Update for workflow

evldn committed:
Update code

evldn committed:
Update code

m-ly4 commented on clang/labs/lab1/volodin_evgeniy/PrintClassPlugin.cpp:
```diff
@@ -0,0 +1,52 @@
+    }
+    return true;
+  }
+  ASTContext *Context;
```
Is it necessary here?

m-ly4 commented on clang/labs/lab1/volodin_evgeniy/PrintClassPlugin.cpp:
```diff
@@ -0,0 +1,52 @@
+  CreateASTConsumer(clang::CompilerInstance &Compiler,
+                    llvm::StringRef InFile) override {
+    return std::unique_ptr<clang::ASTConsumer>(
+        new PrintClassConsumer(Compiler));
```
Could you please make a smart ptr without explicit `new` call?

m-ly4 commented on clang/labs/lab1/volodin_evgeniy/PrintClassPlugin.cpp:
```diff
@@ -0,0 +1,52 @@
+
+  bool ParseArgs(const CompilerInstance &CI,
+                 const std::vector<std::string> &args) override {
+    return true;
```
Could you please add and test `help` option for the plugin?

evldn committed:
1. The ASTContext *Context field was removed from the PrintClassVisitor class as unnecessary.
2. The CreateASTConsumer method of the PrintClassLASTAction class now returns make_unique, not new.
3. The help option for the plugin has been added and tested.

evldn commented:
The following changes have been made:
1. The ASTContext *Context field was removed from the PrintClassVisitor class as unnecessary.
2. The CreateASTConsumer method of the PrintClassLASTAction class now returns make_unique, not new.
3. The help option for the plugin has been added and tested.

evldn committed:
Update code for reload actions (windows-build failed).
